packages/react-native-sdk/src/hooks/useMutingState.ts

```diff
@@ -2,18 +2,42 @@ import {
   useStreamVideoStoreSetState,
   useStreamVideoStoreValue,
 } from '../contexts/StreamVideoContext';
+import {
+  isCameraPermissionGranted$,
+  isMicPermissionGranted$,
+} from '../utils/StreamVideoRN/permissions';
 /**
  * A helper hook which exposes audio, video mute and camera facing mode and
  * their respective functions to toggle state
  *
  * @category Device Management
  */
 export const useMutingState = () => {
-  const isAudioMuted = useStreamVideoStoreValue((store) => store.isAudioMuted);
-  const isVideoMuted = useStreamVideoStoreValue((store) => store.isVideoMuted);
+  const isAudioMutedState = useStreamVideoStoreValue(
+    (store) => store.isAudioMuted,
+  );
+  const isVideoMutedState = useStreamVideoStoreValue(
+    (store) => store.isVideoMuted,
+  );
   const setState = useStreamVideoStoreSetState();
-  const toggleAudioState = () => setState({ isAudioMuted: !isAudioMuted });
-  const toggleVideoState = () => setState({ isVideoMuted: !isVideoMuted });
+  const isMicPermissionGranted = isMicPermissionGranted$.getValue();
+  const isCameraPermissionGranted = isCameraPermissionGranted$.getValue();
+  const toggleAudioState = () => {
+    if (!isMicPermissionGranted) {
+      console.warn('Microphone permission not granted');
```

Comment:
can we do descriptive warnings here 

example.. Microphone permission is not granted. Hence cannot get your audio for calls.

Reply:
Yep ;)